pkg/maps/cpumap/config.go

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+package cpumap
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/hive/cell"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/datapath/xdp"
+)
+
+// userCfg allows for users to specify configuration options on
+// the command line.
+type userCfg struct {
+	XdpCpumapQSize uint32
+}
+
+var defaultUserConfig = userCfg{
+	// TODO: This is picked based off of some initial testing, but
+	// it should probably be tuned based off of the system's specs.
+	XdpCpumapQSize: 255,
+}
+
+func (u userCfg) Flags(flags *pflag.FlagSet) {
+	xdpCPUMapQSizeFlagName := "xdp-cpumap-qsize"
+	flags.Uint32(xdpCPUMapQSizeFlagName, u.XdpCpumapQSize, "Redirection queue size for the XDP cpumap.")
+}
+
+// Config represents the materialized cpumap configuration to be
+// used, depending on its required use by other features.
+type Config struct {
+	enabled bool
+	numCPUs uint
+	qsize   uint32
+}
+
+// If the cpumap is enabled or not.
+func (c Config) Enabled() bool { return c.enabled }
+
+// The number of CPUs that are enabled on the node.
+func (c Config) NumCPUs() uint { return c.numCPUs }
```

Comment:
Technically this is the number of possible CPUs on the node, not the number of enabled CPUs on the node, right? In particular, I'm reminded of issues we had in the past when the two values were confused: https://github.com/cilium/cilium/pull/12121

Reply:
Ah good point. TIL about CPU hot-swapping. Thanks!

Reply:
So this value is set [here](https://github.com/cilium/cilium/pull/35661/files#diff-e3c20aac539bc98432fb973e781277d771069860a2fda58fa0857fdaf4ec8201R104). It used to use `GetNumPossibleCPUs` and if an error was returned, fall back to `runtime.NumCPU`. I changed it to just `NumCPU`.

I'm 90% sure that the map must be set with the number of actual CPUs on the node rather than the number of possible CPUs. The map is used to perform packet steering from one CPU to another and assumes that each entry in the map is a valid redirection target, therefore if the map size is not equal to the number of CPUs available to perform redirection we'll start seeing packet drops.

Thanks for pointing out that `__NR_CPUS__` is set to the number of possible CPUs and not the number of actual CPUs. TIL. I'll need to add a new macro which plumbs the actual number of CPUs into the datapath for this map.

Reply:
Thanks for confirming! Yeah I think it makes sense then to introduce a new macro for the datapath. 

It might also make sense to think about what would happen if more CPUs are attached (this is probably fine, we're just not using them), but also what happens if CPUs are removed. To be honest, I'm not even sure if the rest of Cilium deals gracefully with this.

Reply:
Hm that's a good point. If CPUs are removed then there would be packet drops, since the XDP program would try to redirect to a CPU that isn't there anymore. I could add a warning in the docs about this to make it clear. Maybe we could come up with a better solution as a follow-up?

Reply:
Yes, I absolutely think this is something to tackle in a follow-up. As I've said, I'm not even sure if the rest of Cilium deals with this correctly either. So this would need some experimentation for sure.